ambry-router/src/test/java/com.github.ambry.router/GetManagerTest.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.commons.ByteBufferReadableStreamChannel;
+import com.github.ambry.commons.LoggingNotificationSystem;
+import com.github.ambry.config.RouterConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.messageformat.BlobInfo;
+import com.github.ambry.messageformat.BlobProperties;
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class GetManagerTest {
+  private final MockServerLayout mockServerLayout;
+  private final MockTime mockTime = new MockTime();
+  private final MockClusterMap mockClusterMap;
+  // this is a reference to the state used by the mockSelector. just allows tests to manipulate the state.
+  private AtomicReference<MockSelectorState> mockSelectorState = new AtomicReference<MockSelectorState>();
+  private NonBlockingRouter router;
+
+  private int chunkSize;
+  private int requestParallelism;
+  private int successTarget;
+  private final Random random = new Random();
+
+  // Request params;
+  BlobProperties putBlobProperties;
+  byte[] putUserMetadata;
+  byte[] putContent;
+  ReadableStreamChannel putChannel;
+
+  private static final int MAX_PORTS_PLAIN_TEXT = 3;
+  private static final int MAX_PORTS_SSL = 3;
+  private static final int CHECKOUT_TIMEOUT_MS = 1000;
+
+  /**
+   * Pre-initialization common to all tests.
+   */
+  public GetManagerTest()
+      throws Exception {
```

Comment:
Does it make a difference? The caller of this method does not really care. It just makes development easier if the test methods throw `Exception`.